### PR TITLE
Make `wasm-builder` print the rustc version

### PR DESCRIPTION
**Pull request by bkchr**
This makes `wasm-builder` print the rustc version that is being used to
compile the project. This is rather useful, because people can check
faster if the used rustc version is maybe known for being broken with
Substrate.

### CODE REVIEW

File: utils/wasm-builder/src/wasm_project.rs [hunk 1]
```diff
@@ -34,6 +34,15 @@ use fs2::FileExt;
 
 use itertools::Itertools;
 
+/// Print an info message.
+fn print_info_message(message: &str) -> String {
```
Comment (pepyakin) [inline utils/wasm-builder/src/wasm_project.rs:38]: That doesn't seem to be printing anything. Consider either renaming this method or adding format args and make it actually handling printing.

Comment (pepyakin) [reply utils/wasm-builder/src/wasm_project.rs:38]: Ah, I noticed this method is used for formating an message for an `Err`. Then I guess renaming is the least hassle.

How about, idk, `colorize`?

Comment (bkchr) [inline utils/wasm-builder/src/wasm_project.rs:38]:
```suggestion
/// Colorize an info message.
///
/// Returns the colorized message.
fn colorize_info_message(message: &str) -> String {
```

File: utils/wasm-builder/src/wasm_project.rs [hunk 2]
```diff
@@ -476,7 +486,9 @@ fn build_project(project: &Path, default_rustflags: &str) {
 		build_cmd.arg("--release");
 	};
 
-	println!("Executing build command: {:?}", build_cmd);
+	println!("{}", print_info_message("Information that should be included in a bug report."));
+	println!("{} {:?}", print_info_message("Executing build command:"), build_cmd);
+	println!("{} {}", print_info_message("Using rustc version:"), cargo_cmd.rustc_version());
```
Comment (bkchr) [inline utils/wasm-builder/src/wasm_project.rs:491]:
```suggestion
	println!("{}", colorize_info_message("Information that should be included in a bug report."));
	println!("{} {:?}", colorize_info_message("Executing build command:"), build_cmd);
	println!("{} {}", colorize_info_message("Using rustc version:"), cargo_cmd.rustc_version());
```

File: utils/wasm-builder/src/wasm_project.rs (hunk 1 above)
Comment (bkchr) [reply utils/wasm-builder/src/wasm_project.rs]: Done

File: utils/wasm-builder/src/prerequisites.rs [hunk 3]
```diff
@@ -31,61 +33,101 @@ fn print_error_message(message: &str) -> String {
...
+		"#
+	);
+	write_file_if_changed(
+		main_rs_file,
+		r#"
+			fn main() {
+				println!("{}", env!("RUSTC_VERSION"));
+			}
+		"#
+	);
```
Comment (pepyakin) [inline utils/wasm-builder/src/prerequisites.rs:98]: That's rather a sophisticated way to ask the version? Out of curiosity, why we do it like that compared to just invoking `rustc --version` or something alike?

Comment (bkchr) [reply utils/wasm-builder/src/prerequisites.rs:98]: The best solution would be `cargo rustc --version`, but that doesn't work. I want to know the rust version that is being used by the cargo command that I figured out is using nightly.

TLDR: There is no other way to get the rustc version I'm aware of.

Comment (pepyakin) [reply utils/wasm-builder/src/prerequisites.rs:98]: ah, thanks for the clarification. Mind leaving a comment in code in posterity?

**Review by andresilva — APPROVED** [review]
